examples/sinatra/.toys.rb

```diff
@@ -0,0 +1,192 @@
+  desc "Send an HTTP request to a running Functions Framework"
+
+  flag :https, desc: "Send request using https"
+  flag :host, default: "localhost", desc: "The host to send request to (defaults to localhost)"
+  flag :port, accept: ::Integer, default: 8080, desc: "The port to listen on (defaults to 8080)"
+  flag :path, default: "", desc: "URL path"
+
+  include :exec, e: true
+
+  def run
+    puts "Sending HTTP request"
+    scheme = https ? "https" : "http"
+    response = capture ["curl", "--silent", "#{scheme}://#{host}:#{port}/#{path}"]
+    puts "Response: #{response.inspect}"
+  end
+end
+
+tool "image" do
+  desc "Tools related to running the framework in a local Docker container"
+
+  tool "build" do
+    desc "Build the functions into a local Docker image"
+
+    flag :image, default: "functions-framework-sinatra-test", desc: "The Docker image name"
+    flag :use_release, desc: "Use the released functions_framework gem instead of the local source"
+
+    include "shared"
+    include :exec, e: true
+
+    def run
+      vendor_framework !use_release
+      ::Dir.chdir context_directory
+      exec ["docker", "build", "--tag", image, "."]
+    end
+  end
+
+  tool "server" do
+    desc "Run the locally built Docker image"
+
+    flag :image, default: "functions-framework-sinatra-test", desc: "The Docker image name"
+    flag :port, accept: ::Integer, default: 8080, desc: "The port to listen on (defaults to 8080)"
+    flag :target, accept: ::String, default: "sinatra_example", desc: "The name of the function to serve (defaults to sinatra_example)"
+
+    include :exec, e: true
+
+    def run
+      exec ["docker", "run",
+            "--rm", "-it", "-p", "#{port}:#{port}",
+            image, "--port", port.to_s, "--target", target]
+    end
+  end
+end
+
+tool "run" do
+  desc "Tools related to running the framework in Cloud Run"
+
+  tool "deploy" do
+    desc "Deploy the functions to Cloud Run"
+
+    flag :project, accept: ::String, desc: "The project ID (defaults to the gcloud default project)"
+    flag :app_name, default: "sinatra", desc: 'Name of the Cloud Run app (defaults to "sinatra")'
+    flag :tag, accept: ::String, desc: "Docker tag used as a build ID (defaults to current timestamp)"
+    flag :use_release, desc: "Use the released functions_framework gem instead of the local source"
+    flag :target, accept: ::String, default: "sinatra_example", desc: "The name of the function to serve (defaults to sinatra_example)"
+
+    include "shared"
+    include :exec, e: true
+
+    def run
+      app_tag = tag || default_tag
+      app_project = project || default_project
+      image = "gcr.io/#{app_project}/#{app_name}:#{app_tag}"
+      vendor_framework !use_release
+      ::Dir.chdir context_directory
+      exec ["gcloud", "builds", "submit", "--tag", image, "."]
+      exec ["gcloud", "run", "deploy", app_name,
+            "--image", image,
+            "--platform", "managed",
+            "--allow-unauthenticated",
```

Comment:
Is it possible to _opt in_ for allowing unauthentication instead of _opt out_?
Cloud Run is meant to be secure by default.

Reply:
Having to obtain credentials to send requests to an example, seemed not to be what we wanted.

Reply:
This is an easy way to add auth to cURL:
https://gist.github.com/grant/ab0c5e1e68580ed403871318c4da0396